### PR TITLE
Npgsql 6 upgrade

### PR DESCRIPTION
**Pull request by oskardudycz**
- Updated Weasel to 2.0.0 and fixed DateTime related tests to follow new Npgsql rules https://www.npgsql.org/doc/index.html (DateTimeOffset for UTC, DateTime for without time zone).
- Added NodaTime mappings.

See related PR for Weasel: https://github.com/JasperFx/weasel/pull/35

### CODE REVIEW

**Comment by jeremydmiller** [comment]
You already know this, but that's a 5.0 release. I think it's smart to replace the DateTime parameters w/ DateTimeOffset. I'm good with everything I see here.

**Review by jeremydmiller — APPROVED** [review]
I think this is good as is

File: src/Marten.NodaTime/NodaTimeExtensions.cs [hunk 1]
```diff
@@ -18,30 +21,63 @@ public static class NodaTimeExtensions
...
-                storeOptions.Serializer(serializer);
+                    break;
+                default:
+                    throw new NotSupportedException("Current serializer cannot be automatically configured for NodaTime. Set shouldConfigureJsonSerializer to false if you're using your own serializer.");
             }
+
+            storeOptions.Serializer(serializer);
+        }
+
+        public static void SetNodaTimeMappings()
```
Comment (mysticmind) [inline src/Marten.NodaTime/NodaTimeExtensions.cs:52]: Suggestion: Rename this from `SetNodaTimeMappings` to `SetNodaTimeTypeMappings` to add clarity.

**Review by mysticmind — APPROVED** [review]
Looks good to me. Added a suggestion, please have a look.

File: src/Marten.NodaTime/NodaTimeExtensions.cs (hunk 1 above)
Comment (oskardudycz) [reply src/Marten.NodaTime/NodaTimeExtensions.cs]: Fixed in: https://github.com/JasperFx/marten/pull/1987/commits/380b2b39d57ad6454e1fa770d20a3cb022931268

**Pull request merged**